linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml

```diff
@@ -349,14 +349,14 @@
   loop: "{{ res_def['networks']|product(res_count)|list }}"
   when:
     - res_def['networks'] is defined
-    - net.dhcp is not defined or net.dhcp == true
+    - net[0]['dhcp'] is not defined or net[0]['dhcp']== true
 
 - set_fact:
     macs: []
 
 - set_fact:
-    macs: "{{ macs + [mac_addresses.results[node.1].matches[node.0].mac.address] }}"
-  loop: "{{ range(0,res_def['networks']|length,1)|list|product(res_count)|list }}"
+    macs: "{{ macs + [mac_addresses.results[node].matches[0].mac.address] }}"
```

Comment:
The bug is that in a case where a node has multiple networks but one them had dhcp disabled then linchpin will ignore when looking for mac address (line 352) but will try to get it in when going over each node and its networks(line 359). The solution is not to ignore a case where node can have multiple networks like you did but to skip missing networks, something like:
```
macs: "{{ macs + [mac_addresses.results[node.1].matches[node.0].mac.address | default('')] | difference(['']) }}"
```

Reply:
@abraverm thanks! I added your suggestion.

Reply:
@abraverm Looks like I get the same with the last patch:
```bash
TASK [libvirt : set_fact] ******************************************************
ok: [localhost] => (item=[0, 0]) => {"ansible_facts": {"macs": ["52:54:00:2e:a4:2a"]}, "changed": false, "node": [0, 0]}
ok: [localhost] => (item=[0, 1]) => {"ansible_facts": {"macs": ["52:54:00:2e:a4:2a", "52:54:00:bf:01:98"]}, "changed": false, "node": [0, 1]}
ok: [localhost] => (item=[0, 2]) => {"ansible_facts": {"macs": ["52:54:00:2e:a4:2a", "52:54:00:bf:01:98", "52:54:00:1f:6f:a7"]}, "changed": false, "node": [0, 2]}
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: list object has no element 1\n\nThe error appears to have been in '/tmp/linchpin/.venv/lib/python2.7/site-packages/linchpin/provision/roles/libvirt/tasks/provision_libvirt_node.yml': line 378, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- set_fact:\n  ^ here\n"}
	to retry, use: --limit @/tmp/linchpin/.venv/lib/python2.7/site-packages/linchpin/provision/libvirt.retry
```